addons/mail/static/src/models/user/user.js

```diff
@@ -35,6 +55,70 @@ function factory(dependencies) {
             return partner.nameOrDisplayName;
         }
 
+        /**
+         * Opens a chat between the partner of this user and the current partner
+         * and returns it.
+         *
+         * If a chat already exists for those partners it is reused. Otherwise a
+         * new chat is created.
+         *
+         * If a chat is not appropriate a notification is displayed instead.
+         *
+         * @param {Object} [params]
+         * @returns {mail.thread|undefined}
+         */
+        async openChat(params) {
+            if (!this.partner) {
+                await this.async(() => this.fetchPartner());
+            }
+            if (!this.partner) {
```

Comment:
add comment this can only happen if user was inserted with a bad id, might be worth calling this.delete then